src/main/java/io/codelex/flightplanner/flightplan/FlightPlanService.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FlightPlanService {
+
+    private FlightPlanRepository flightPlanRepository;
+
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    public FlightPlanService(FlightPlanRepository flightPlanRepository) {
+        this.flightPlanRepository = flightPlanRepository;
+
+    }
+
+    public FlightPlanRepository getFlightPlanRepository() {
+        return flightPlanRepository;
+    }
+
+    public void setFlightPlanRepository(FlightPlanRepository flightPlanRepository) {
+        this.flightPlanRepository = flightPlanRepository;
+    }
+
+
+    public void clearFlights() {
+        this.flightPlanRepository.setIdsGiven(0);
+        this.flightPlanRepository.setPlan(new ArrayList<>());
+    }
+
+    public Flight setID(FlightRequest flightRequest) {
+        int newID = this.flightPlanRepository.getIdsGiven().incrementAndGet();
+        return new Flight(newID,
+                flightRequest.getFrom(),
+                flightRequest.getTo(),
+                flightRequest.getCarrier(),
+                flightRequest.getDepartureTime(),
+                flightRequest.getArrivalTime());
+    }
+
+    public Flight addFlight(Flight flight) {
+
+        if (flightAlreadyAdded(flight)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Can't add same flight twice");
+        } else if (sameAirport(flight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Same Airports");
+        } else if (!validDatesFlight(flight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Arrival after departure");
+        } else {
+            this.flightPlanRepository.saveFlight(flight);
+            return flight;
+        }
+    }
+
+    public boolean flightAlreadyAdded(Flight flight) {
+        return this.flightPlanRepository.getPlan().stream()
+                .anyMatch(storedFlight -> storedFlight.equals(flight));
+    }
+
+
+    public boolean sameAirport(Flight flight) {
+        return flight.getFrom().equals(flight.getTo());
+    }
+
+    public boolean validDatesFlight(Flight flight) {
+        return LocalDateTime
+                .parse(flight.getDepartureTime(), dateTimeFormatter)
+                .isBefore(LocalDateTime.parse(flight.getArrivalTime(), dateTimeFormatter));
+
+    }
+
+    public Flight fetchFlight(int id) {
+        Flight found = this.flightPlanRepository.getFlight(id);
+        if (found == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        return found;
+    }
+
+    public void deleteFlight(int id) {
+        this.flightPlanRepository.deleteFlight(id);
+    }
+
+    public Airport[] searchAirport(String phrase) {
```

Comment:
Šī metode ļoti neparasti uzrakstīta un manuprāt būtu noteikti jāpamaina.
Izmanto vienkārši streams un metodi filter, nevis pats iterē cauri un sauc to matches.toArray..
Nav īsti skaidrs kāpēc rezultāti sākumā ir List, bet pēc tam konvertē par Array - kāpēc metode nevar vienkārši atgriezt List<Airport>?
Mans ieteikums - izmanto stream.